lms/djangoapps/courseware/views/index.py

```diff
@@ -152,34 +148,25 @@ def get(self, request, course_id, chapter=None, section=None, position=None):
 
                 self.view = STUDENT_VIEW
 
-                # Do the enrollment check if enable_unenrolled_access is not enabled.
                 self.course = get_course_with_access(
                     request.user, 'load', self.course_key,
                     depth=CONTENT_DEPTH,
-                    check_if_enrolled=not self.enable_unenrolled_access,
+                    check_if_enrolled=True,
+                    check_if_authenticated=True
                 )
                 self.course_overview = CourseOverview.get_from_id(self.course.id)
+                self.is_staff = has_access(request.user, 'staff', self.course)
 
-                if self.enable_unenrolled_access:
-                    # Check if the user is considered enrolled (i.e. is an enrolled learner or staff).
-                    try:
-                        check_course_access(
-                            self.course, request.user, 'load', check_if_enrolled=True,
-                        )
-                    except CourseAccessRedirect as exception:
-                        # If the user is not considered enrolled:
-                        if self.course.course_visibility == COURSE_VISIBILITY_PUBLIC:
-                            # If course visibility is public show the XBlock public_view.
-                            self.view = PUBLIC_VIEW
-                        else:
-                            # Otherwise deny them access.
-                            raise exception
-                    else:
-                        # If the user is considered enrolled show the default XBlock student_view.
-                        pass
+                # There's only one situation where we want to show the public view
+                if (
```

Comment:
(Read comment on 166 in the old code first)

In our new code, we know that by the time we've gotten here, we've already raised any redirects correctly in `get_course_with_access`, taking into account the unenrolled access flag.  

That means in this `if`, we can just focus on identifying conditions under which we want to show the PUBLIC_VIEW.

There's only one of those.

1. The user can't be staff
2. The unenrolled access flag must be on
3. The course_visibility must be public.
4. The user can't be enrolled.

If those four conditions are met, great!  Use PUBLIC_VIEW.  Otherwise we always use STUDENT_VIEW.